Refactorize source and payload factories

poncovka opened the pull request:
Reorganize and extend tests for the source and payload factories
and simplify kickstart processing in the Payloads module.

The changes are consistent with the code of the Storage module
and the partitioning factory.

jkonecny12 commented on pyanaconda/modules/payloads/payloads.py:
```diff
@@ -97,19 +97,18 @@ def process_kickstart(self, data):
 
-        self.set_payload(payload)
-        PayloadContainer.to_object_path(payload)
+        # FIXME: Process packages in the DNF module.
```
Add here please that `Packages` module will be converted to a structure as explanation why.

poncovka replied:
Fixed.

poncovka commented:
Updated. I have renamed only the test file for the Payloads module. We don't really use the service name as a prefix of all related test files. However, we could eventually move these files to new subdirectories.

VladimirSlavik commented:
I agree we should start thinking about the test file organization - some of the "module test" files looked ridiculously large to me.
